docs/migration.md

```diff
@@ -71,6 +75,47 @@ from pydantic.v1.utils import lenient_isinstance
 
 Pydantic V1 documentation is available at [https://docs.pydantic.dev/1.10/](https://docs.pydantic.dev/1.10/).
 
+
+#### Using Pydantic v1 features in a v1/v2 environment
+
+As of `pydantic>=1.10.17` the `pydantic.v1` namespace is available to use within v1,
+meaning that in order to unpin a `pydantic<2` dependency to allow using `v1` features,
+simply replace `pydantic<2` with `pydantic>=1.10.17,<3` and find and replace all:
+
+```python test="skip" lint="skip" upgrade="skip"
+from pydantic.<module> import <object>
+```
+
+to:
+
+```python test="skip" lint="skip" upgrade="skip"
+from pydantic.v1.<module> import <object>
+```
+
+Imports for `pydantic` v1 features for different versions of pydantic.
+
+=== "`pydantic>2,<3`"
+
+    ```python test="skip" lint="skip" upgrade="skip"
+    try:
+        from pydantic.v1.fields import ModelField
+    except ImportError:
+        from pydantic.fields import ModelField
+    ```
```

Comment:
Hmm, I think the versions are off for this case. I think you'd only need this in the case that your v1 version is <1.10.17...

Reply:
yeah its actually a funny one, because its about using v1 features - when you have v2 installed, so if your python version is less than 2 anyway then you wouldn't need this import anyway. I see what you mean though, I think having the bound as `pydantic<3` will give the desired interpretation i.e.:

`pydantic<3` this import style should always work (more cumbersome but more general)
`pydantic>=1.10.17,<3` this import style should work (less cumbersome),
 WDYT?

Reply:
Yep, I think that sounds great.

Reply:
Nice work using the mkdocs material tab feature, this is a great use case for that